cpp/daal/include/algorithms/algorithm_container_base_batch.h

```diff
@@ -143,10 +144,15 @@ class AlgorithmContainerImpl<batch> : public AlgorithmContainer<batch>
  * \tparam avx2Container        Implementation for Intel(R) Advanced Vector Extensions 2 (Intel(R) AVX2)
  * \tparam avx512Container      Implementation for Intel(R) Xeon(R) processors based on Intel AVX-512
  */
+#ifdef __ARM_ARCH
```

Comment:
Start with the x86 check, and have an error condition for unknown architectures. This is a general comment, where a check is done for the (micro-)architecture

Reply:
What about other macros? I believe I've seen at least `__arch64__`, `_M_ARM64` somewhere in ARM specific code (in other repos)